openrouteservice/src/main/java/org/heigit/ors/servlet/listeners/ORSKafkaConsumerInitContextListener.java

```diff
@@ -0,0 +1,97 @@
+/*
+ *  Licensed to GIScience Research Group, Heidelberg University (GIScience)
+ *
+ *   http://giscience.uni-hd.de
+ *   http://heigit.org
+ *
+ *  under one or more contributor license agreements. See the NOTICE file
+ *  distributed with this work for additional information regarding copyright
+ *  ownership. The GIScience licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.heigit.ors.servlet.listeners;
+
+import com.typesafe.config.ConfigException;
+import com.typesafe.config.ConfigObject;
+import org.apache.log4j.Logger;
+import org.heigit.ors.config.AppConfig;
+import org.heigit.ors.kafka.ORSKafkaConsumer;
+import org.heigit.ors.kafka.ORSKafkaConsumerConfiguration;
+import org.heigit.ors.kafka.ORSKafkaTestCluster;
+import org.heigit.ors.routing.RoutingProfileManager;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+import java.util.ArrayList;
+import java.util.List;
+
+@WebListener
+public class ORSKafkaConsumerInitContextListener implements ServletContextListener {
+    private static final Logger LOGGER = Logger.getLogger(ORSKafkaConsumerInitContextListener.class);
+    private ORSKafkaConsumer consumer;
+    private ORSKafkaTestCluster testCluster;
+
+    @Override
+    public void contextInitialized(ServletContextEvent contextEvent) {
+        while (!RoutingProfileManager.isInitCompleted()) { // wait until ORS init is completed
+            try {
+                Thread.sleep(1000);
```

Comment:
Can we always be sure that this object is invoked from a thread different from the main thread? Otherwise infinite loop, isnt it? Idk

Reply:
servlet listeners are executed in the order they are listen in the `web.xml`, so the `ORSInitContextListener` is run before `ORSKafkaConsumerInitContextListener` and starts intialization. Tweaked the `isInitComplete()` method anyway so that it won't be a problem even if we were to change up that order in the future for whatever reason.